superset/assets/javascripts/dashboard/v2/containers/DashboardGrid.jsx

```diff
@@ -7,11 +7,18 @@ import {
   resizeComponent,
 } from '../actions/dashboardLayout';
 
+function mapStateToProps({ dashboard }, ownProps) {
+  return {
+    editMode: dashboard.editMode,
+    cells: ownProps.cells,
```

Comment:
why are you overwriting cells with the same `ownProps`? also see my comments about removing cells.